### PR TITLE
Remove photutils from Astroquery astrometry.net

### PR DESCRIPTION
**Pull request by mfixstsci**
This PR addresses #2782.

Currently `astroquery/astrometry_net/core.py` tries to import `photutils.detection.DAOStarFinder` and `astropy.nddata.CCDData` and will make decisions on how sources in the fits file passed to `AstrometryNetClass` are extracted. The following additions will prepare the photutils portion of the algorithm for removal with the goal that if a source catalog is not provided, that `nova.astrometry.net` will perform the source extraction.

- Added astropy deprecation warning in code block containing `photutils`.
- Added astropy deprecation decorator for and unused parameter and parameter that allows user to bypass or allow `nova.astrometry.net` to extract sources.

Fixes: #2782.

### CODE REVIEW

File: astroquery/astrometry_net/core.py [hunk 1]
```diff
@@ -412,6 +415,11 @@ def solve_from_image(self, image_file_path, *, force_image_upload=False,
                                          cache=False,
                                          files={'file': f})
         else:
+            warning_msg = "Removing photutils functionality to obtain extracted positions list from " \
+                          "AstoromertyNetClass.solve_from_source_list. Users will need to " \
```
Comment (keflavich) [inline astroquery/astrometry_net/core.py:419]:
```suggestion
                          "AstrometryNetClass.solve_from_source_list. Users will need to " \
```

Comment (mfixstsci) [reply astroquery/astrometry_net/core.py]: Hey @keflavich thank you for the feedback. I fixed the typo in [4e23354](https://github.com/astropy/astroquery/pull/3067/commits/4e2335417ecc2d04c319e071422742b47846c28b)

**Comment by bsipocz** [comment]
@mfixstsci - Tip of the day, use these words to cross-link the issue and thus autoclose. Unfortunately 'address' is not part of the list:

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by mfixstsci** [comment]
> @mfixstsci - Tip of the day, use these words to cross-link the issue and thus autoclose. Unfortunately 'address' is not part of the list:
> 
> https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

Thank you this is really useful!

File: astroquery/astrometry_net/core.py [hunk 2]
```diff
@@ -412,6 +415,11 @@ def solve_from_image(self, image_file_path, *, force_image_upload=False,
                                          cache=False,
                                          files={'file': f})
         else:
+            warning_msg = "Removing photutils functionality to obtain extracted positions list from " \
+                          "AstrometryNetClass.solve_from_source_list. Users will need to " \
+                          "submit pre-extracted catalog positions or a fits file for https://nova.astrometry.net/ " \
+                          "to extract with their algorithm."
```
Comment (bsipocz) [inline astroquery/astrometry_net/core.py:421]: Nitpick, but I would rephrase this a bit, as it's the user who sees this message not a third party.
```suggestion
            warning_msg = ("Running photutils functionality within astroquery to obtain extracted positions list from " 
                          "AstrometryNetClass.solve_from_source_list() is deprecated and will be removed in the next release. "
                          "You need to submit catalog positions or a fits file to obtain an astrometric solution using this service.")
```

**Review by bsipocz — COMMENTED** [review]
Nitpick, otherwise it all looks good. 

Thanks @mfixstsci

**Review by bsipocz — APPROVED** [review]
Thanks @mfixstsci!
